File: client/java/src/main/java/io/be/mesh/struct/Outbound.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package io.be.mesh.struct;

import io.be.mesh.macro.Index;
import io.be.mesh.mpc.Returns;
import lombok.Data;

import java.io.Serializable;

/**
 * @author coyzeng@gmail.com
 */
@Data
public class Outbound implements Serializable, Returns {

    private static final long serialVersionUID = 4142652099532835971L;

    @Index(0)
    private String code;

    @Index(5)
    private String message;

    @Index(10)
    private Cause cause;

    @Index(15)
    private Object content;

}

File: client/python/mesh/kinds/outbound.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
from typing import Any

from mesh.macro import index, serializable


@serializable
class Outbound:

    @index(0)
    def code(self) -> str:
        return ''

    @index(1)
    def message(self) -> str:
        return ''

    @index(2)
    def cause(self) -> str:
        return ''

    @index(3)
    def content(self) -> Any:
        return None
